opm/parser/eclipse/EclipseState/Schedule/Group/Group2.hpp

```diff
@@ -23,15 +23,91 @@
 
 #include <string>
 
+#include <opm/parser/eclipse/EclipseState/Util/IOrderSet.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
 
 namespace Opm {
 
 class Group2 {
 public:
-    explicit Group2(const std::string& group_name);
 
+struct GroupInjectionProperties {
+    Phase phase = Phase::WATER;
+    GroupInjection::ControlEnum cmode = GroupInjection::NONE;
+    double surface_max_rate = 0;
+    double resv_max_rate = 0;
+    double target_reinj_fraction = 0;
+    double target_void_fraction = 0;
+
+    bool operator==(const GroupInjectionProperties& other) const;
+    bool operator!=(const GroupInjectionProperties& other) const;
+};
+
+
+struct GroupProductionProperties {
+    GroupProduction::ControlEnum cmode = GroupProduction::NONE;
+    GroupProductionExceedLimit::ActionEnum exceed_action = GroupProductionExceedLimit::NONE;
+    double oil_target = 0;
+    double water_target = 0;
+    double gas_target = 0;
+    double liquid_target = 0;
+    double resv_target = 0;
+
+    bool operator==(const GroupProductionProperties& other) const;
+    bool operator!=(const GroupProductionProperties& other) const;
+};
+
+    Group2(const std::string& group_name, std::size_t insert_index_arg, std::size_t init_step_arg);
+
+    bool defined(std::size_t timeStep) const;
+    std::size_t insert_index() const;
+    const std::string& name() const;
+    const GroupProductionProperties& productionProperties() const;
+    const GroupInjectionProperties& injectionProperties() const;
+    int getGroupNetVFPTable() const;
+    bool updateNetVFPTable(int vfp_arg);
+    bool update_gefac(double gefac, bool transfer_gefac);
+
+    const std::string& parent() const;
+    bool updateParent(const std::string& parent);
+    bool updateInjection(const GroupInjectionProperties& injection);
+    bool updateProduction(const GroupProductionProperties& production);
```

Comment:
I think you should add a comment that explains the return value of `update*()`.  It is not at all obvious to me what either of `false` or `true` mean here.

Reply:
This pattern applies to many classes in the `Schedule`. The `updateXXXX()` methods return true if whatever internal property has been updated; i.e. in a fictious class containing an integer:

```
class my_int {
public:
     bool update(int new_value) {
          if (this->value == new_value)
                 return;
          this->value = new_value;
          return true;
     }
private:
    int value;
}
```
These classes are then typically owned by a `DynamicState<T>` - and the return value from the `updateXXX()` determines whether a new instance should be installed in the `DynamicState`. I am happy to add comment - but maybe rather in the `Schedule.hpp` file?

Reply:
> I am happy to add comment - but maybe rather in the `Schedule.hpp` file?

Sounds good.